pkg/metallb/frr.go

```diff
@@ -0,0 +1,664 @@
+		}
+
+		builder.Object, err = builder.convertToStructured(unsObject)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return builder, err
+}
+
+// Get returns FRRConfiguration object if found.
+func (builder *FrrConfigurationBuilder) Get() (*frrtypes.FRRConfiguration, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof(
+		"Collecting FRRConfiguration object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	unsObject, err := builder.apiClient.Resource(
+		GetFrrConfigurationGVR()).Namespace(builder.Definition.Namespace).Get(
+		context.TODO(), builder.Definition.Name, metav1.GetOptions{})
+
+	if err != nil {
+		glog.V(100).Infof(
+			"Failed to pull an unstructured object in namespace %s",
+			builder.Definition.Namespace)
+
+		return nil, err
+	}
+
+	return builder.convertToStructured(unsObject)
+}
+
+// Delete removes FrrConfigurationBuilder object from a cluster.
+func (builder *FrrConfigurationBuilder) Delete() (*FrrConfigurationBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the FrrConfiguration object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("FrrConfiguration cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Resource(
+		GetFrrConfigurationGVR()).Namespace(builder.Definition.Namespace).Delete(
+		context.TODO(), builder.Definition.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete FRRConfiguration: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// GetFrrConfigurationGVR returns FrrConfiguration GroupVersionResource which could be used for Clean function.
+func GetFrrConfigurationGVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group: FRRAPIGroup, Version: APIVersion, Resource: "frrconfigurations",
+	}
+}
+
+// Exists checks whether the given FrrConfiguration exists.
+func (builder *FrrConfigurationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
+		"Checking if FrrConfiguration %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithBGPRouter defines a BGP router as localASN.
+func (builder *FrrConfigurationBuilder) WithBGPRouter(localASN uint32) *FrrConfigurationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Define BGP router with an ASN: %v", localASN)
+
+	// Check if there are no routers defined
+	if len(builder.Definition.Spec.BGP.Routers) == 0 {
+		// Create the first router
+		builder.Definition.Spec.BGP.Routers = []frrtypes.Router{
+			{
+				ASN: localASN,
+			},
+		}
+	} else {
```

Comment:
you can append to an empty (or nil) slice so this block is unnecessary

Reply:
If I remember correctly I was getting out of range panic errors. I will try again.

Reply:
Made the change with no issue of panic.